roachtest: use enum for cloud value

RaduBerinde opened the pull request:
This commit changes cloud values to an enum instead of a more
error-prone string.

Epic: none
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/127126)
<!-- Reviewable:end -->

RaduBerinde committed:
roachtest: use enum for cloud value

This commit changes cloud values to an enum instead of a more
error-prone string.

Epic: none
Release note: None

RaduBerinde commented:
Here's a pointer to a case where the bad string led to a test not running for a long time: https://github.com/cockroachdb/cockroach/issues/126881#issuecomment-2220051022

RaduBerinde commented:
Friendly ping

srosenberg commented on pkg/cmd/roachtest/registry/testdata/filter/errors:
```diff
@@ -1,9 +1,5 @@
 
-filter cloud=badcloud
-----
-error: invalid cloud "badcloud"; valid clouds are local,gce,aws,azure
```
Not quite sure why this test case was removed?

RaduBerinde replied:
TFTR! This code that this tests now takes a `spec.Cloud`. It's no longer technically possible to pass an arbitrary cloud string.

srosenberg reviewed (approved):
Thanks, LGTM! Let's smoke test before merging.

renatolabs reviewed (approved):
> Here's a pointer to a case where the bad string led to a test not running for a long time: https://github.com/cockroachdb/cockroach/issues/126881#issuecomment-2220051022

FWIW, the test was not being skipped because of a bad string; it was being skipped because both the GCS and AWS variants were being declared with the `CompatibleClouds` field set to `AllExceptAWS`.

This is a good change to make regardless, thank you!

RaduBerinde commented:
TFTRs!

> Thanks, LGTM! Let's smoke test before merging.

How can I set the test probability these days? I no longer see it under Parameters

renatolabs commented:
If you just run the nightly build on this branch without any special parameters, it should automatically run with a probability of 0.1; that is already > 100 tests and should be sufficient to verify that the changes here don't break anything.

RaduBerinde commented:
Thanks!
https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16125125
https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAwsBazel/16125126

RaduBerinde commented:
bors r+

RaduBerinde commented:
bors r-

RaduBerinde commented:
bors r+
